internal/pkg/manifest/svc.go

```diff
@@ -148,6 +148,30 @@ func (c *Count) UnmarshalYAML(value *yaml.Node) error {
 	return nil
 }
 
+// UnmarshalYAML overrides the default YAML unmarshaling logic for the ScalingConfigOrPercentage
+// struct, allowing it to perform more complex unmarshaling behavior.
+// This method implements the yaml.Unmarshaler (v3) interface.
+func (r *ScalingConfigOrPercentage) UnmarshalYAML(value *yaml.Node) error {
+	if err := value.Decode(&r.ScalingConfig); err != nil {
+		switch err.(type) {
+		case *yaml.TypeError:
+			break
+		default:
+			return err
+		}
+	}
+
+	if !r.ScalingConfig.IsEmpty() {
+		// Successfully unmarshalled ScalingConfig fields, return
+		return nil
+	}
+
+	if err := value.Decode(&r.Value); err != nil {
+		return errUnmarshalCountOpts
```

Comment:
```suggestion
		return errors.New(`unable to unmarshal into int or composite-style map`))
```
Since this is not unmarshalling the `count` field 😛